pkg/services/live/push/push.go

```diff
@@ -58,25 +80,97 @@ func (g *Gateway) IsEnabled() bool {
 	return g.Cfg.IsLiveEnabled() // turn on when Live on for now.
 }
 
-func (g *Gateway) Handle(ctx *models.ReqContext) {
-	streamID := ctx.Params(":streamId")
+func extractKey(req *http.Request) string {
+	header := req.Header.Get("Authorization")
+	parts := strings.SplitN(header, " ", 2)
+	var keyString string
+	if len(parts) == 2 && parts[0] == "Bearer" {
+		keyString = parts[1]
+	}
+	return keyString
+}
+
+func (g *Gateway) isAuthenticated(req *http.Request) (bool, error) {
```

Comment:
We already have similar code here: https://github.com/grafana/grafana/blob/438b403acc3e4a374872104a2ef1a510abdd5075/pkg/services/contexthandler/contexthandler.go#L140-L192

Possible to create something that can be reused by both to decrease code duplication?

Reply:
Noticed the comment below now. Thinks that a good idea to do either now or slightly after this has been merged

Reply:
Yes, planning to do this inside this pr